File: odbc_tests/tests/e2e/tls/crl_enabled.cpp
#include <catch2/catch_test_macros.hpp>

#include "../../../common/include/Connection.hpp"

// Scenario: Should connect and select with CRL enabled
TEST_CASE("Should connect and select with CRL enabled") {
  // Given Snowflake client is logged in
  auto connection_string = get_connection_string();
  // And CRL is enabled
  connection_string += "CRL_MODE=ENABLED;";

  // When Query "SELECT 1" is executed
  Connection conn(connection_string);
  auto stmt = conn.execute_fetch("SELECT 1");

  // Then the request attempt should be successful
  SQLLEN value = 0;
  SQLGetData(stmt.getHandle(), 1, SQL_C_SLONG, &value, 0, nullptr);
  REQUIRE(value == 1);
}

File: sf_core/tests/e2e/tls/crl_enabled.rs
use crate::common::snowflake_test_client::SnowflakeTestClient;

#[test]
fn should_connect_and_select_with_crl_enabled() {
    // Given Snowflake client is configured before connect
    let client = SnowflakeTestClient::with_default_jwt_auth_params();

    // And CRL is enabled before the connection is established
    client.set_connection_option("crl_mode", "ENABLED");

    // Given Snowflake client is logged in
    let connection_result = client.connect();
    connection_result.expect("Login failed");

    // When Query "SELECT 1" is executed
    let result = client.execute_query("SELECT 1");

    // Then the request attempt should be successful
    let rows = crate::common::arrow_result_helper::ArrowResultHelper::from_result(result)
        .transform_into_array::<i64>()
        .unwrap();
    assert_eq!(rows.len(), 1);
    assert_eq!(rows[0][0], 1);
}
